### DISCUSSION TITLE
Backtick-delimited strings for quote-intense strings (i.e. JSON)

### DISCUSSION THREAD

**Discussion by AndreasHassing** [dotnet/csharplang#4222] [Language Ideas]
Example (like JavaScript's string interpolation):

```csharp
var document =
  `{ "hello": "world", "age": "${myAge}" }`;
```

Note that it would be nice if string interpolation in this mode is default and needs to have a `$` prefix for the LHS `{`, to allow declaring JSON cleanly in code.

**Comment by gfoidl** [comment]
> Tilde-delimited

Where is the tilde `~` in your proposal?

But you're right that it types nicer than `$"{{ 'hello': 'world', 'age': '{myAge}' }}"`.

**Reply by TheUnlocked** [reply]
My guess is that they're mixing up the names for tilde and backtick since they're on the same key on a qwerty keyboard.

**Reply by AndreasHassing** [reply]
Sorry, fixed. Meant backtick-delimited 😄

**Comment by HaloFour** [comment]
I don't think the suggested requirement of having the interpolated expressions be prefixed with `$` would fly as it would be very confusing to require that in one form of interpolation but not any others and it would cause the following two expressions to result in different strings:

```cs
var arg = "bar";
var str1 = $`foo: ${arg}`;
var str2 = $"foo: ${arg}";
```

**Reply by AndreasHassing** [reply]
Changed the suggestion, such that string interpolation is enabled by default (and verbatim?), but requires `${}`. That helps, right?

**Comment by leoniDEV** [comment]
I really hate when the backtick is used, simply because there is no backtick on my keyboard.
There are many keyboard layout that does not have the backtick, so you have to switch the keyboard language in the OS configs or use the ASCII code, these two things are not impossible to do but they are not so practical so I always prefer other form of delimitation

**Reply by AndreasHassing** [reply]
I understand your frustration. On Danish keyboard, it's not easy to get the backtick into the editor, which is one of the reasons I am emulating US keyboard on top of the Danish-specific keys.

Language-specific QWERTY-like keyboards are not inherently good for programming.

**Comment by huoyaoyuan** [comment]
Generally, using interpolation to declare json is not a good idea. It's very likely to cause escaping problems.

**Reply by AndreasHassing** [reply]
Can you elaborate?

Devs using C# are declaring JSON in code and it's very hard to decode using existing constructs.
Importing a file and filling in placeholders is prone to runtime failures which is worse IMO.

**Reply by huoyaoyuan** [reply]
When I need json, I always create an object and serialize it. 
In your sample, `myAge` is an integer. When it's a string, it's definitely not a good idea to use interpolation.

**Reply by TahirAhmadov** [reply]
I'd agree, in general, creating strings in code like this has a smell to it, JSON or otherwise. JSON should be serialized to a TextWriter, to avoid buffering; labels should be creating using `string.Format`, because then the formatting string can be substituted easily for localization; etc.
